pkg/util/ec2/ec2_tags.go

```diff
@@ -29,6 +30,46 @@ var (
 )
 
 func fetchEc2Tags(ctx context.Context) ([]string, error) {
+	if config.Datadog.GetBool("collect_ec2_tags_use_imds") {
+		// prefer to fetch tags from IMDS, falling back to the API
+		tags, err := fetchEc2TagsFromIMDS(ctx)
+		if err == nil {
+			return tags, nil
+		}
+
+		log.Debugf("Could not fetech tags from instance metadata (trying EC2 API instead): %s", err)
+	}
+
+	tags, err := fetchEc2TagsFromAPI(ctx)
+	if err == nil {
+		return tags, err
+	}
+
+	return nil, err
```

Comment:
Why not using return `fetchEc2TagsFromAPI(ctx)`?

Reply:
Ah, that would be simpler :)